veps/sig-network/persistent-mac-addresses.md

```diff
@@ -0,0 +1,313 @@
+
+A new MAC synchronizer will be added to the existing VM controller in `virt-controller`, following the same pattern as the firmware UUID synchronizer. The synchronizer implements the `synchronizer` interface with a `Sync(vm, vmi)` method that is called during the VM controller's reconciliation loop.
+
+The MAC synchronizer is invoked automatically during VM reconciliation when a VMI exists. The synchronizer checks if the VMI is in `Running` phase with populated `status.interfaces`. For each interface in the VM template spec that has an empty `macAddress` field, it finds the matching interface in VMI status (by name) and constructs a JSON patch to add the MAC address. All patches are applied in a single API call to the VM object.
+
+The synchronizer is idempotent - if MAC addresses are already present in the VM spec, it skips them and returns immediately. User-specified MACs are never modified. The synchronizer runs in the same reconciliation loop as other VM synchronizers (network, firmware, instancetype), requiring no additional watches or event handlers.
+
+### MAC Address Source
+
+MAC addresses are read from `vmi.Status.Interfaces[].MAC`, which is already reliably populated by:
+1. libvirt domain specification, originating from CNI-allocated pod network interfaces
+2. guest-agent, when exists.
+
+This VEP also suggests to populate the MAC from multus-status, as stated later in the SR-IOV Persistence case.
+
+### Idempotency and Safety
+
+- Controller only patches VM spec if `macAddress` field is currently empty
+- Once persisted, subsequent reconciliations skip that interface (no-op)
+- User-specified MACs are never overwritten
+- Safe to call multiple times (defense against controller restarts, race conditions)
+- Interface matching by name ensures correct MAC is persisted to correct interface
+
+### Edge Cases Handled
+
+**SR-IOV MAC Persistence:**
+Currently, SR-IOV status MAC is only [populated](https://github.com/kubevirt/kubevirt/blob/59c62130ead7e3b05b3de737f71aba3aae58e798/pkg/network/setup/netstat.go#L307) when explicitly set in VM spec.
+So in case of SR-IOV, a prerequisite step would take place to extract the MAC address from the pod's `network-status` annotation into `vmi.Status.Interfaces[].MAC`, following the existing pattern for IPs and DeviceInfo extraction. The MAC synchronizer will then persist it to VM spec as normal.
+
+**VMI Recreation:**
+- On VM restart, if MACs already persisted, they are reused (no new allocation)
+- If MACs not yet persisted, waits for new VMI to reach Running state
+
+**Controller Restarts:**
+- Idempotent design allows safe reconciliation after controller restart
+
+**Active Migrations:**
+- Synchronizer skips persisting MACs when VMI is an active migration source 
+- This prevents persisting the source VMI's MAC when it's about to be replaced by the target VMI
+- After migration completes, the target VMI's MAC is persisted (ensuring the correct, surviving MAC is preserved)
```

Comment:
This is mainly describing a cross-namespace (or cluster) migration, which is a special case.
In a regular live migration, there is only one single VMI.

 I guess this can be generalized so the synchronizer will just avoid the reflection of the mac during any live migration, which should in practice never happen, because the mac is reflected on the status once the domain is created and that is expected to happen before or in parallel to the domain being detected as "running".
 
 I would prefer this would be just a backup safety measurement, but I am also fine to keep it.